engine/src/main/resources/org/camunda/bpm/engine/impl/mapping/entity/HistoricDetail.xml

```diff
@@ -505,6 +505,7 @@
     <if test="taskProcessInstanceIds != null &amp;&amp; taskProcessInstanceIds.size > 0">
       inner join ${prefix}ACT_HI_TASKINST T
         on TASK_ID_ = T.ID_
+        and TASK_ID_ IS NOT NULL
```

Comment:
❓ Does this "collide" in any way with the same clause we already use in one of the consumers of this SQL snippet? Tests look fine but I don't know if we actually test those preconditions together in any way.